arangod/Aql/IndexExecutor.cpp

```diff
@@ -450,6 +450,15 @@ IndexExecutor::IndexExecutor(Fetcher& fetcher, Infos& infos)
       _currentIndex(_infos.getIndexes().size()),
       _skipped(0) {
   TRI_ASSERT(!_infos.getIndexes().empty());
+
+  if (!_infos.getEngine()->waitForSatellites(_infos.getCollection())) {
+    double maxWait = _infos.getEngine()->getQuery()->queryOptions().satelliteSyncWait;
+    THROW_ARANGO_EXCEPTION_MESSAGE(TRI_ERROR_CLUSTER_AQL_COLLECTION_OUT_OF_SYNC,
+                                   "collection " + _infos.getCollection()->name() +
+                                   " did not come into sync in time (" +
+                                   std::to_string(maxWait) + ")");
+  }
```

Comment:
Wondering why all this is not handled inside `waitForSatellites(...)`, and why `waitForSatellites` doesn't get the max wait time as a parameter.
Then `waitForSatellites(...)` could handle the waiting and throw an exception in case the collection didn't get in sync timely enough. Also meaning it doesn't need to return a bool value.